mkdocs/docs/reference/tasks/metrics.md

```diff
@@ -0,0 +1,93 @@
+---
+template: main.html
+---
+
+# Metrics Tasks
+
+## `metrics/collect`
+
+### Overview
+
+The `metrics/collect` takes enables collection of [builtin metrics](). It generates a stream of HTTP requests to one or more app/ML model versions, and collects latency/error metrics.
+
+### Example
+
+The following loop action contains a `metrics/collect` task which is executed during every loop of the experiment. The task sends a stream of HTTP requests to the version named `sample-app-v1` at the specified URL and collects builtin latency/error metrics for this version.
+
+```yaml
+loop:
+- task: metrics/collect
+  with:
+    versions: 
+    - name: sample-app-v1
+      url: http://sample-app.default.svc.cluster.local
+```
+
+### Inputs
+
+<!-- const (
+	// CollectTaskName is the name of the task this file implements
+	CollectTaskName string = "collect"
+
+	// DefaultQPS is the default value of QPS (queries per sec) in collect task inputs
+	DefaultQPS float32 = 8
+
+	// DefaultTime is the default value of time (duration of queries) in collect task inputs
+	DefaultTime string = "5s"
+)
+
+// Version contains header and url information needed to send requests to each version.
+type Version struct {
+	// name of the version
+	// version names must be unique and must match one of the version names in the
+	// VersionInfo field of the experiment
+	Name string `json:"name" yaml:"name"`
+	// how many queries per second will be sent to this version; optional; default 8
+	QPS *float32 `json:"qps,omitempty" yaml:"qps,omitempty"`
+	// HTTP headers to use in the query for this version; optional
+	Headers map[string]string `json:"headers,omitempty" yaml:"headers,omitempty"`
+	// URL to use for querying this version
+	URL string `json:"url" yaml:"url"`
+}
+
+// CollectInputs contain the inputs to the metrics collection task to be executed.
+type CollectInputs struct {
+	// how long to run the metrics collector; optional; default 5s
+	Time *string `json:"time,omitempty" yaml:"time,omitempty"`
+	// list of versions
+	Versions []Version `json:"versions" yaml:"versions"`
+	// URL of the JSON file to send during the query; optional
+	PayloadURL *string `json:"payloadURL,omitempty" yaml:"payloadURL,omitempty"`
+	// if LoadOnly is set to true, this task will send requests without collecting metrics; optional
+	LoadOnly *bool `json:"loadOnly,omitempty" yaml:"loadOnly,omitempty"`	
+} -->
+
+| Field name | Field type | Description | Required |
+| ----- | ---- | ----------- | -------- |
+| time | string | Duration of the `metrics/collect` task run. Specified in the [Go duration string format](https://golang.org/pkg/time/#ParseDuration). Default value is `5s`. | No |
+| payloadURL | string | URL of JSON-encoded data. If this field is specified, the metrics collector will send HTTP POST requests to versions, and the POST requests will contain this JSON data as payload. | No |
```

Comment:
This says _payloadURL_. The example has _url_. 
How does this work for the bookinfo app that requires a Host header also encoded in the fortio request?

Reply:
@kalantar Those are two separate fields! 

URL is a sub-field within the version field. It represents the URL where  the version serves requests. It is mandatory.

payloadURL is an optional field (top level). KFServing and Seldon examples require POSTing JSON to targets. `payloadURL` enables you to specify a location for this JSON payload.